### PR TITLE
feat: Add requesting API name to permission prompt

### PR DESCRIPTION
**Pull request by bartlomieju**
<s>Blocked on https://github.com/denoland/deno/pull/15907</s>

Adds information which API requested certain permission.

### CODE REVIEW

File: runtime/js/40_spawn.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
   const promiseIdSymbol = SymbolFor("Deno.core.internalPromiseId");
 
-  function spawnChild(command, {
+  function spawnChildInner(command, {
```
Comment (Phosra) [inline runtime/js/40_spawn.js:24]: Missing an `apiName` parameter?

Comment (bartlomieju) [reply runtime/js/40_spawn.js:24]: How so? It's just a helper function that's called by other functions

Comment (Phosra) [reply runtime/js/40_spawn.js:24]: On line 55 within this file, this function (`spawnChildInner`) is called with three arguements; `spawnChild` is a wrapper around `spawnChildInner`, yet `spawnChildInner` doesn't use the API name passed to it?

**Comment by bartlomieju** [comment]
I think this is good to go for now. I still need to add these hints to FFI APIs, but I'd prefer to do that in a separate PR.

File: runtime/permissions.rs [hunk 2]
```diff
@@ -2084,6 +2186,9 @@ fn permission_prompt(message: &str, name: &str) -> bool {
   eprint!("{}", colors::bold("Deno requests "));
   eprint!("{}", colors::bold(message));
   eprintln!("{}", colors::bold("."));
+  if let Some(api_name) = api_name {
```
Comment (bartlomieju) [inline runtime/permissions.rs:2189]: Need to update count of lines to clear in case this var is non-empty

File: runtime/js/40_spawn.js [hunk 3]
```diff
@@ -204,7 +208,7 @@
       if (this.#rid === null) {
         throw new TypeError("Child process has already terminated.");
       }
-      ops.op_kill(this.#pid, signo);
+      ops.op_kill(this.#pid, signo, "Child.kill()");
```
Comment (crowlKats) [inline runtime/js/40_spawn.js:211]:
```suggestion
      ops.op_kill(this.#pid, signo, "Deno.Child.kill()");
```

**Review by dsherret — APPROVED** [review]
LGTM

File: runtime/js/40_process.js [hunk 4]
```diff
@@ -91,7 +95,7 @@
     }
 
     kill(signo) {
-      opKill(this.pid, signo);
+      opKill(this.pid, signo, "Process.kill()");
```
Comment (crowlKats) [inline runtime/js/40_process.js:98]:
```suggestion
      opKill(this.pid, signo, "Deno.Process.kill()");
```

File: ext/net/ops.rs [hunk 5]
```diff
@@ -253,7 +253,7 @@ where
       {
         let mut s = state.borrow_mut();
         s.borrow_mut::<NP>()
-          .check_net(&(&args.hostname, Some(args.port)))?;
+          .check_net(&(&args.hostname, Some(args.port)), "Datagram.send()")?;
```
Comment (crowlKats) [inline ext/net/ops.rs:256]:
```suggestion
          .check_net(&(&args.hostname, Some(args.port)), "Deno.DatagramConn.send()")?;
```

File: ext/net/ops.rs [hunk 6]
```diff
@@ -278,7 +278,8 @@ where
       let address_path = Path::new(&args.path);
       {
         let mut s = state.borrow_mut();
-        s.borrow_mut::<NP>().check_write(address_path)?;
+        s.borrow_mut::<NP>()
+          .check_write(address_path, "Datagram.send()")?;
```
Comment (crowlKats) [inline ext/net/ops.rs:282]:
```suggestion
          .check_write(address_path, "Deno.DatagramConn.send()")?;
```

**Review by crowlKats — CHANGES_REQUESTED** [review]

**Review by crowlKats — APPROVED** [review]
LGTM

**Pull request merged**